### PR TITLE
Jwttoken update

### PR DESCRIPTION
**Pull request by deedee**

### CODE REVIEW

File: src/java/main/com/topcoder/direct/services/view/interceptors/AuthenticationInterceptor.java [hunk 1]
```diff
@@ -282,30 +283,30 @@ public String intercept(ActionInvocation invocation) throws Exception {
...
         if (jwtCookie == null) {
             return loginPageName;
         }
 
-        Map<String, Object> decodedPayload;
-
+        JWTToken jwtToken = null;
         try {
-            decodedPayload = new JWTVerifier(DirectProperties.CLIENT_SECRET_AUTH0, DirectProperties.CLIENT_ID_AUTH0).verify(jwtCookie.getValue());
+            String[] knownIssuers = new String[]{ "https://" + DirectProperties.DOMAIN_AUTH0 };
```
Comment (skyhit) [inline src/java/main/com/topcoder/direct/services/view/interceptors/AuthenticationInterceptor.java]: @deedee we used a different domain, like topcoder-dev.auth0.com, topcoder-newauth0.auth0.com, can you make it configurable through tokens.properties ?

File: src/java/main/com/topcoder/direct/services/view/processor/security/LoginProcessor.java [hunk 2]
```diff
@@ -75,7 +75,7 @@
 
     static {
         JWT_OPTIONS = new DirectJWTSigner.Options();
-        JWT_OPTIONS.setAlgorithm(Algorithm.HS256);
+        JWT_OPTIONS.setAlgorithm(Algorithm.HMAC256(DirectProperties.CLIENT_SECRET_AUTH0.getBytes()));
```
Comment (skyhit) [inline src/java/main/com/topcoder/direct/services/view/processor/security/LoginProcessor.java]: @deedee why change this? will it backward compatible?

Comment (deedee) [reply src/java/main/com/topcoder/direct/services/view/processor/security/LoginProcessor.java]: New version of library is not backward compatible. So we need some adjustments regards on this

File: src/java/main/com/topcoder/direct/services/view/interceptors/AuthenticationInterceptor.java (hunk 1 above)
Comment (deedee) [reply src/java/main/com/topcoder/direct/services/view/interceptors/AuthenticationInterceptor.java]: New token.properties added: ```@JWT_VALID_ISSUERS@```. Named all issuers here, including issuer for old jwt (comma separate)

File: token.properties.example [hunk 3]
```diff
@@ -394,6 +394,7 @@
 @REG_SERVER_NAME@=tc.cloud.topcoder.com
 @LDAP_AUTH0_CONNECTION_NAME@=vm-ldap-connection
 @member.profile.url.base@=http://tc.cloud.topcoder.com
+@JWT_VALID_ISSUERS@ = https://sma.auth0.com, https://newtc.auth0.com
```
Comment (skyhit) [inline token.properties.example]: @deedee if you didn't set `https://topcoder-newauth.auth0.com`, will you pass the test? or it is purposely, you didn't put the correct value?

Comment (deedee) [reply token.properties.example]: It will not pass  . You need to list all issuers here

**Pull request merged**